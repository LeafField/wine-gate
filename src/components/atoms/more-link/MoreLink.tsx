import React, { FC } from "react";
import Link from "next/link";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowRightLong } from "@fortawesome/free-solid-svg-icons";

type Props = {
  link: string;
};

const MoreLink: FC<Props> = ({ link }) => {
  return (
    <Link
      href={`/category/${link}`}
      className="mt-2 flex items-center justify-end text-black transition-colors duration-300 hover:text-blue"
      prefetch={false}
    >
      <span className="text-main-text">もっと見る</span>
      <FontAwesomeIcon
        className="h-[w-[clamp(0.875rem,1.54vw,1rem)] ml-[0.125rem] w-[clamp(0.875rem,1.54vw,1rem)]"
        icon={faArrowRightLong}
      />
    </Link>
  );
};

export default MoreLink;
