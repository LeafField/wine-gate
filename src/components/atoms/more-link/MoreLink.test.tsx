import { screen, render } from "@testing-library/react";
import MoreLink from "./MoreLink";

describe("MoreLinkの単体テスト", () => {
  test("MoreLinkコンポーネントが存在することを確認する", () => {
    const { container } = render(<MoreLink link="sober" />);
    expect(screen.getByText("もっと見る")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("propsのlinkを元にリンクが正しく生成されているか", () => {
    render(<MoreLink link="sober" />);
    expect(screen.getByRole("link")).toHaveAttribute(
      "href",
      "/category/sober/new/1",
    );
  });
});
