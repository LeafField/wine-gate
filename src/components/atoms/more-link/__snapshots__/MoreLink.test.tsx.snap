// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreLinkの単体テスト MoreLinkコンポーネントが存在することを確認する 1`] = `
<div>
  <a
    class="mt-2 flex items-center justify-end text-black transition-colors duration-300 hover:text-blue"
    href="/category/sober"
  >
    <span
      class="text-main-text"
    >
      もっと見る
    </span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-arrow-right-long h-[w-[clamp(0.875rem,1.54vw,1rem)] ml-[0.125rem] w-[clamp(0.875rem,1.54vw,1rem)]"
      data-icon="arrow-right-long"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;
